tools/options.c

```diff
@@ -345,7 +345,9 @@ static void nn_append_string (struct nn_parse_context *ctx,
     if (!lst->items) {
         nn_memory_error (ctx);
     }
-    lst->items[lst->num-1] = str;
```

Comment:
How about just adding `nn_assert (lst && lst->items);`? I agree the logic change obfuscates.